src/Program.cs

```diff
@@ -0,0 +1,59 @@
+﻿using CommandLine;
+using Newtonsoft.Json;
+using System.Net;
+using IP2Azure;
+
+Parser.Default.ParseArguments<Options>(args)
+    .WithParsed<Options>(opts =>
+        {
+            IPAddress? ip = null;
+            try {
+                ip = IPAddress.Parse(opts.Address);
+            }
+            catch (FormatException)
+            {
+                Console.WriteLine($"{opts.Address} is not a valid IPV4 address.");
+                Environment.Exit(1);
+            }
+            string json = System.IO.File.ReadAllText(opts.AzureJson);
+
+            // Map of IP Range (as string) to a list of services.
+            var servicesMap = JsonConvert.DeserializeObject<ServiceRanges>(json)?.GetServicesByRange();
+
+            if (servicesMap is null) {
+                Console.WriteLine($"Could not parse the JSON file {opts.AzureJson}.");
```

Comment:
Do you think it would make sense to use `Console.Error.WriteLine` here (and for other diagnostic messages) to make parsing the output a bit easier?

Reply:
Done, thanks!